docs/administration-guide/gaffer-deployment/gremlin.md

```diff
@@ -0,0 +1,164 @@
+# Connecting Gremlin to Gaffer
+
+It is possible to use Gremlin as an alternative querying language in Gaffer. To
+make Gremlin available however, there are some additional steps that need to be
+taken to connect to a Gaffer graph via this interface.
+
+## Overview
+
+Gremlin serves as a query layer for a graph that implements the Tinkerpop graph
+structure. As of v2.1.0 Gaffer has made available a library that can be utilised
+to enable Gremlin queries. This library can be included via maven in any project
+using the following dependency definition:
+
+```xml
+<dependency>
+    <groupId>uk.gov.gchq.gaffer</groupId>
+    <artifactId>tinkerpop</artifactId>
+    <version>${gaffer.version}</version>
+</dependency>
+```
+
+The library contains the graph implementation that allows Tinkerpop to talk to a
+Gaffer graph and generally is all that is needed to provide the functionality.
+To actually utilise Gremlin queries however, a connection to what's known as a
+`GraphTraversalSource` is required which is the class from which Gremlin
+traversals are spawned. To do this we recommend utilising the provided
+[Gremlin server framework](https://tinkerpop.apache.org/docs/current/reference/#connecting-gremlin-server)
+which can be configured to use the Gaffer Tinkerpop implementation so that a
+endpoint is available for Gremlin queries.
+
+!!! note
+    For release of Gaffer v2.2.0 a ready made container image will be provided
+    to run a preconfigured Gremlin server for Gaffer.
+
+## Connecting to Any Existing Gaffer Graph
+
+The simplest way to connect Gremlin to an existing Gaffer instance where you may
+not know the Store type or Schema would be via a [Proxy Store](../gaffer-stores/proxy-store.md).
+Connecting this way means Gremlin communicates via the Gaffer REST API
+(similar to [gafferpy](../../user-guide/apis/python-api.md)) meaning there may
+be a performance hit for larger queries.
+
+!!! tip
+    You can also of course connect directly to an existing instance's storage
+    layer too (e.g. Accumulo store) but this would require a more complex
+    configuration and knowledge of the Schema.
+
+The general connection diagram looks something like the following:
+
+```mermaid
+flowchart LR
+    A(["User"])
+    --> B("Gremlin Server")
+    --> C(Gaffer Proxy Store)
+    --> D(Existing Gaffer Instance)
+```
+
+To establish this connection three configuration files are required:
+
+- `store.properties` - The Gaffer store configuration for the Proxy Store.
```

Comment:
Can we specify if not already where these files need to live please?

Reply:
They can live anywhere in theory, its hopefully clarified later that they just need bind mounting into the container at run time (the paths there are given).